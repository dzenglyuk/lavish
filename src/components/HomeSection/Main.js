import React from "react";
import Button from "../Button/Main";
import { Link } from "react-router-dom";
import "./Main.css";

const HomeSection = ({
    lightBg,
    topLine,
    lightText,
    lightTextDesc,
    headLine,
    description,
    buttonLabel,
    img,
    alt,
    imgStart,
}) => {
    return (
        <div className={lightBg ? "home__section" : "home__section darkBg"}>
            <div className="container">
                <div
                    className="row home__home-row"
                    style={{
                        display: "flex",
                        flexDirection:
                            imgStart === "start" ? "row-reverse" : "row",
                    }}
                >
                    <div className="col">
                        <div className="home__text-wrapper">
                            <div className="top-line">{topLine}</div>
                            <h1 className={lightText ? "heading" : "heading dark"}>{headLine}</h1>
                            <p className={lightTextDesc ? "home__subtitle" : "home__subtitle dark"}> {description} </p>
                            <Link to="/sign-up">
                                <Button btnSize="btn--wide" btnColor="blue">{buttonLabel}</Button>
                            </Link>
                        </div>
                    </div>
                    <div className="col">
                        <div className="home__img-wrapper">
                            <img src={img} alt={alt} className="home__img"/>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default HomeSection;
