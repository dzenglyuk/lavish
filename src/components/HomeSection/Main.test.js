import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";

import {
    homeObjOne,
    homeObjTwo,
    homeObjThree,
    homeObjFour,
} from "../../pages/Data";
import HomeSection from "./Main";

test("Renders correctly", () => {
    const { getByText, getByAltText } = render(
        <Router>
            <HomeSection {...homeObjOne} />
        </Router>
    );
    getByText(homeObjOne.headLine);
    getByText(homeObjOne.topLine);
    getByText(homeObjOne.description);
    getByText(homeObjOne.buttonLabel);
    getByAltText(homeObjOne.alt);
});

test("Matches the snapshot 1", () => {
    const tree = renderer
        .create(
            <Router>
                <HomeSection {...homeObjOne} />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 2", () => {
    const tree = renderer
        .create(
            <Router>
                <HomeSection {...homeObjTwo} />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 3", () => {
    const tree = renderer
        .create(
            <Router>
                <HomeSection {...homeObjThree} />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 4", () => {
    const tree = renderer
        .create(
            <Router>
                <HomeSection {...homeObjFour} />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
