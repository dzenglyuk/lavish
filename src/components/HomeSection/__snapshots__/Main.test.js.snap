// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1 1`] = `
<div
  className="home__section darkBg"
>
  <div
    className="container"
  >
    <div
      className="row home__home-row"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="home__text-wrapper"
        >
          <div
            className="top-line"
          >
            Exclusive Access
          </div>
          <h1
            className="heading"
          >
            Unlimited Transactions with zero fees
          </h1>
          <p
            className="home__subtitle"
          >
             
            Get access to our exclusive diamond card that allows you to send unlimited transactions without getting charged any fees
             
          </p>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            <button
              className="btn 
			btn--primary
			btn--wide
			blue"
            >
              Get Started
            </button>
          </a>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="home__img-wrapper"
        >
          <img
            alt="Credit Card"
            className="home__img"
            src="images/svg-1.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Matches the snapshot 2 1`] = `
<div
  className="home__section"
>
  <div
    className="container"
  >
    <div
      className="row home__home-row"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="home__text-wrapper"
        >
          <div
            className="top-line"
          >
            EASY SETUP
          </div>
          <h1
            className="heading dark"
          >
            Super fast and simple onboarding process
          </h1>
          <p
            className="home__subtitle dark"
          >
             
            Get everything set up and ready in under 10 minutes. All you need to do is add your information and you're ready to go.
             
          </p>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            <button
              className="btn 
			btn--primary
			btn--wide
			blue"
            >
              Start Now
            </button>
          </a>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="home__img-wrapper"
        >
          <img
            alt="Vault"
            className="home__img"
            src="images/svg-7.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Matches the snapshot 3 1`] = `
<div
  className="home__section darkBg"
>
  <div
    className="container"
  >
    <div
      className="row home__home-row"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="home__text-wrapper"
        >
          <div
            className="top-line"
          >
            100% SECURE
          </div>
          <h1
            className="heading"
          >
            Stay protected 24/7 anywhere anytime
          </h1>
          <p
            className="home__subtitle"
          >
             
            We have you covered no matter where you are located. Over 140 locations worldwide to ensure you have access anytime
             
          </p>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            <button
              className="btn 
			btn--primary
			btn--wide
			blue"
            >
              Learn More
            </button>
          </a>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="home__img-wrapper"
        >
          <img
            alt="Vault"
            className="home__img"
            src="images/svg-5.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Matches the snapshot 4 1`] = `
<div
  className="home__section darkBg"
>
  <div
    className="container"
  >
    <div
      className="row home__home-row"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="home__text-wrapper"
        >
          <div
            className="top-line"
          >
            DATA ANALYTICS
          </div>
          <h1
            className="heading"
          >
            Every transaction is stored on our secure cloud database
          </h1>
          <p
            className="home__subtitle"
          >
             
            Never ever have to worry again about saved reciepts. We store your data, so you can access it anytime.
             
          </p>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            <button
              className="btn 
			btn--primary
			btn--wide
			blue"
            >
              Sign Up Now
            </button>
          </a>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="home__img-wrapper"
        >
          <img
            alt="Vault"
            className="home__img"
            src="images/svg-8.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
